source/core/Ky.ts

```diff
@@ -145,14 +149,14 @@ export class Ky {
 		};
 
 		const isRetriableMethod = ky._options.retry.methods.includes(ky.request.method.toLowerCase());
-		const result = isRetriableMethod ? ky._retry(fn) : fn();
+		const result = (isRetriableMethod ? ky._retry(fn) : fn()) as ResponsePromise;
 
-		for (const [type, mimeType] of Object.entries(responseTypes)) {
-			// @ts-expect-error not sure how to properly type this!
+		for (const [type, mimeType] of Object.entries(responseTypes) as ObjectEntries<typeof responseTypes>) {
 			result[type] = async () => {
 				// eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
 				ky.request.headers.set('accept', ky.request.headers.get('accept') || mimeType);
 
+				// eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion, @typescript-eslint/non-nullable-type-assertion-style
```

Comment:
Why are you silencing the listing? If it's valid, it would be useful with a code comment about why.

Reply:
No idea here, somehow `npm run test` throws an error here

Reply:
Removed it 🔥